src/ui/CategoryFacet/CategoryFacetSearch.ts

```diff
@@ -1,3 +1,199 @@
-// export class CategoryFacetSearch {
-// constructor() {}
-// }
+import { CategoryFacet } from './CategoryFacet';
+import { FacetSearchElement } from '../Facet/FacetSearchElement';
+import { debounce, last, first } from 'underscore';
+import { $$, Dom } from '../../utils/Dom';
+import { SVGDom } from '../../utils/SVGDom';
+import { SVGIcons } from '../../utils/SVGIcons';
+import { l } from '../../strings/Strings';
+import { IGroupByValue } from '../../rest/GroupByValue';
+import { KEYBOARD } from '../../utils/KeyboardUtils';
+import 'styling/_CategoryFacetSearch';
+
+export class CategoryFacetSearch {
+  public container: Dom | undefined;
+  private facetSearchElement: FacetSearchElement;
+  private currentlyDisplayedResults: HTMLElement;
+  private displayNewValues: () => void;
+  private currentResult: Dom | undefined;
+
+  constructor(private categoryFacet: CategoryFacet) {
+    this.facetSearchElement = new FacetSearchElement();
+    this.displayNewValues = debounce(() => {
```

Comment:
I'd move all of that in a separate function instead of directly in the constructor, just for readability.